util/src/logutil.cpp

```diff
@@ -49,22 +70,75 @@ void logFormatter(
   }
 }
 
+void fileLogFormatter(
+  const boost::log::record_view& rec, boost::log::formatting_ostream& strm)
+{
+  auto severity = rec[boost::log::trivial::severity];
+
+  std::string sLevel = boost::log::trivial::to_string(severity.get());
+  std::transform(sLevel.begin(), sLevel.end(), sLevel.begin(), ::toupper);
+
+  strm << "[" << sLevel << "] " << rec[boost::log::expressions::smessage];
 }
 
+} // namespace
+
 boost::log::trivial::severity_level getSeverityLevel()
 {
  return boost::log::attribute_cast<
    boost::log::attributes::mutable_constant<boost::log::trivial::severity_level>>(
      boost::log::core::get()->get_global_attributes()["Severity"]).get();
 }
 
-void initLogger()
+void initConsoleLogger()
 {
+  boost::log::add_common_attributes();
   auto fsSink = boost::log::add_console_log(
     std::cout,
     boost::log::keywords::auto_flush = true);
 
-  fsSink->set_formatter(&logFormatter);
+  fsSink->set_formatter(&consoleLogFormatter);
+}
+
+std::string getLoggingBase(const std::string& path_, const std::string& name_)
+{
+  if (path_.find('~') != std::string::npos)
+  {
+    throw std::invalid_argument("The path should not contain a \'~\' character. \
+                                 Please provide an absolute path");
+  }
+
+  using namespace boost::filesystem;
+
+  boost::system::error_code ec;
+  create_directory(path_, ec);
+  if (ec)
+  {
+    throw std::invalid_argument("Permission denied to create " + path_);
+  }
+
+  return canonical(absolute(path(path_))).string() + '/' + name_ + '_';
```

Comment:
If the `log-target` is `/`, the this path will be `//projectname_parser.log` for the parser, which writes the file in the working directory.

Reply:
It is now possible to use '/'.